controllers/rcj_soccer_referee_supervisor/referee/referee.py

```diff
@@ -0,0 +1,49 @@
+from referee.consts import (
+    GOAL_X_LIMIT,
+    TIME_STEP,
+)
+from referee.supervisor import RCJSoccerSupervisor
+
+
+class RCJSoccerReferee(RCJSoccerSupervisor):
+    def check_robots_in_penalty_area(self):
+        """Check whether robots are violating rule not to stay in
+        penalty area for longer period of time"""
+        # TODO: decide if this rule will be applied and then implement it
+        pass
+
+    def check_goal(self):
+        """Check if goal is scored"""
+        # ball in the blue goal
+        if self.ball_translation[0] > GOAL_X_LIMIT:
+            self.score_yellow += 1
+            self.draw_scores(self.score_blue, self.score_yellow)
+            self.ball_reset_timer = self.post_goal_wait_time
+
+        # ball in the yellow goal
+        elif self.ball_translation[0] < -GOAL_X_LIMIT:
+            self.score_blue += 1
+            self.draw_scores(self.score_blue, self.score_yellow)
+            self.ball_reset_timer = self.post_goal_wait_time
+
+    def tick(self):
+        self.time -= TIME_STEP / 1000.0
+        if self.time < 0:
+            self.time = self.match_time
+
+        self.draw_time(self.time)
+
+        # If we are currently not in the post-goal waiting period,
+        # check if a goal took place, setup the waiting period and move the
+        # robots to proper positions afterwards.
+        if self.ball_reset_timer == 0:
+            self.check_goal()
+            self.check_robots_in_penalty_area()
```

Comment:
@Adman I guess we would do this in the `else` part of the statement

Reply:
Isn't the `else` part for post-goal waiting period where the goal took place?

Reply:
Nah, you are right @Adman, I didn't get it right away. Please ignore my comment here.